pkg/mongo/testing/changestream.go

```diff
@@ -24,28 +24,33 @@ import (
 	"go.mongodb.org/mongo-driver/bson"
 )
 
+var (
+	// nextFnNotExecuted is set so that Next returns true only once, then false, for each test.
+	nextFnNotExecuted = true
+)
+
```

Comment:
this should be part of the TestCSData struct and not a global variable.
In general is more readable if you ask for the positive case rather than the negative one...  in other words, nextFnExecuted instead of nextFnNotExecuted